rs/consensus/src/consensus/validator.rs

```diff
@@ -990,6 +990,20 @@ impl Validator {
                 continue;
             }
 
+            // If there exists an equivocation proof for the signer of the
+            // proposal at the same height, we ignore the block. The only
+            // way in which we would validate such a proposal is through
+            // the notarization fast-path.
```

Comment:
What would happen if we're late and the block was actually finalized? We will eventually fetch both the notification and finalization because we never purge full notarizations, right? cc @eichhorl

Reply:
We do purge notarizations, but only by CUP height (minus `MINIMUM_CHAIN_LENGTH`). So even if the block was finalized, we'll generally be able to get the notarization. 

But that's the case no matter if a block is equivocating or not. If we want to validate any finalization, we need to have the associated notarization in our validated pool.

Reply:
We also purge non-finalized notarizations and blocks below and at the finalized height